// gitlab/dist/es5/templates/ResourceIssueBoards.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _urlJoin = _interopRequireDefault(require("url-join"));

var _infrastructure = require("../infrastructure");

var ResourceIssueBoards =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(ResourceIssueBoards, _BaseService);

  function ResourceIssueBoards(resourceType) {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ResourceIssueBoards);

    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ResourceIssueBoards)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.url = (0, _urlJoin.default)(_this.url, resourceType);
    return _this;
  }

  (0, _createClass2.default)(ResourceIssueBoards, [{
    key: "all",
    value: function all(resourceId, options) {
      var rId = encodeURIComponent(resourceId);
      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/boards"), options);
    }
  }, {
    key: "create",
    value: function create(resourceId, name) {
      var rId = encodeURIComponent(resourceId);
      return _infrastructure.RequestHelper.post(this, "".concat(rId, "/boards"), {
        name: name
      });
    }
  }, {
    key: "createList",
    value: function createList(resourceId, boardId, labelId) {
      var _map = [resourceId, boardId].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          rId = _map2[0],
          bId = _map2[1];

      return _infrastructure.RequestHelper.post(this, "".concat(rId, "/boards/").concat(bId, "/lists"), {
        labelId: labelId
      });
    }
  }, {
    key: "edit",
    value: function edit(resourceId, boardId, options) {
      var _map3 = [resourceId, boardId].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          rId = _map4[0],
          bId = _map4[1];

      return _infrastructure.RequestHelper.put(this, "".concat(rId, "/boards/").concat(bId), options);
    }
  }, {
    key: "editList",
    value: function editList(resourceId, boardId, listId, position) {
      var _map5 = [resourceId, boardId, listId].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 3),
          rId = _map6[0],
          bId = _map6[1],
          lId = _map6[2];

      return _infrastructure.RequestHelper.put(this, "".concat(rId, "/boards/").concat(bId, "/lists/").concat(lId), {
        position: position
      });
    }
  }, {
    key: "lists",
    value: function lists(resourceId, boardId) {
      var _map7 = [resourceId, boardId].map(encodeURIComponent),
          _map8 = (0, _slicedToArray2.default)(_map7, 2),
          rId = _map8[0],
          bId = _map8[1];

      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/boards/").concat(bId, "/lists"));
    }
  }, {
    key: "remove",
    value: function remove(resourceId, boardId) {
      var _map9 = [resourceId, boardId].map(encodeURIComponent),
          _map10 = (0, _slicedToArray2.default)(_map9, 2),
          rId = _map10[0],
          bId = _map10[1];

      return _infrastructure.RequestHelper.delete(this, "".concat(rId, "/boards/").concat(bId));
    }
  }, {
    key: "removeList",
    value: function removeList(resourceId, boardId, listId) {
      var _map11 = [resourceId, boardId, listId].map(encodeURIComponent),
          _map12 = (0, _slicedToArray2.default)(_map11, 3),
          rId = _map12[0],
          bId = _map12[1],
          lId = _map12[2];

      return _infrastructure.RequestHelper.delete(this, "".concat(rId, "/boards/").concat(bId, "/lists/").concat(lId));
    }
  }, {
    key: "show",
    value: function show(resourceId, boardId) {
      var _map13 = [resourceId, boardId].map(encodeURIComponent),
          _map14 = (0, _slicedToArray2.default)(_map13, 2),
          rId = _map14[0],
          bId = _map14[1];

      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/boards/").concat(bId));
    }
  }, {
    key: "showList",
    value: function showList(resourceId, boardId, listId) {
      var _map15 = [resourceId, boardId, listId].map(encodeURIComponent),
          _map16 = (0, _slicedToArray2.default)(_map15, 3),
          rId = _map16[0],
          bId = _map16[1],
          lId = _map16[2];

      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/boards/").concat(bId, "/lists/").concat(lId));
    }
  }]);
  return ResourceIssueBoards;
}(_infrastructure.BaseService);

var _default = ResourceIssueBoards;
exports.default = _default;
