// gitlab/dist/es5/templates/ResourceBadges.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _urlJoin = _interopRequireDefault(require("url-join"));

var _infrastructure = require("../infrastructure");

var ResourceBadges =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(ResourceBadges, _BaseService);

  function ResourceBadges(resourceType) {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ResourceBadges);

    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ResourceBadges)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.url = (0, _urlJoin.default)(_this.url, resourceType);
    return _this;
  }

  (0, _createClass2.default)(ResourceBadges, [{
    key: "add",
    value: function add(resourceId, options) {
      var rId = encodeURIComponent(resourceId);
      return _infrastructure.RequestHelper.post(this, "".concat(rId, "/badges"), options);
    }
  }, {
    key: "all",
    value: function all(resourceId, options) {
      var rId = encodeURIComponent(resourceId);
      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/badges"), options);
    }
  }, {
    key: "edit",
    value: function edit(resourceId, badgeId, options) {
      var _map = [resourceId, badgeId].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          rId = _map2[0],
          bId = _map2[1];

      return _infrastructure.RequestHelper.put(this, "".concat(rId, "/badges/").concat(bId), options);
    }
  }, {
    key: "preview",
    value: function preview(resourceId, linkUrl, imageUrl) {
      var rId = encodeURIComponent(resourceId);
      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/badges/render"), {
        linkUrl: linkUrl,
        imageUrl: imageUrl
      });
    }
  }, {
    key: "remove",
    value: function remove(resourceId, badgeId) {
      var _map3 = [resourceId, badgeId].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          rId = _map4[0],
          bId = _map4[1];

      return _infrastructure.RequestHelper.delete(this, "".concat(rId, "/badges/").concat(bId));
    }
  }, {
    key: "show",
    value: function show(resourceId, badgeId) {
      var _map5 = [resourceId, badgeId].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 2),
          rId = _map6[0],
          bId = _map6[1];

      return _infrastructure.RequestHelper.get(this, "".concat(rId, "/badges/").concat(bId));
    }
  }]);
  return ResourceBadges;
}(_infrastructure.BaseService);

var _default = ResourceBadges;
exports.default = _default;
