// gitlab/dist/es5/templates/ResourceVariables.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _infrastructure = require("../infrastructure");

var url = function url(resourceType, resourceId, resource2Type, resource2Id) {
  var _map = [resourceId, resource2Id].map(encodeURIComponent),
      _map2 = (0, _slicedToArray2.default)(_map, 2),
      rId = _map2[0],
      r2Id = _map2[1];

  var output = "".concat(resourceType, "/").concat(rId, "/");

  if (resource2Id) {
    output += "".concat(resource2Type, "/").concat(r2Id, "/");
  }

  output += 'variables';
  return output;
};

var ResourceVariables =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(ResourceVariables, _BaseService);

  function ResourceVariables(resourceType, resource2Type) {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, ResourceVariables);

    for (var _len = arguments.length, args = new Array(_len > 2 ? _len - 2 : 0), _key = 2; _key < _len; _key++) {
      args[_key - 2] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(ResourceVariables)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.resourceType = resourceType;
    _this.resource2Type = resource2Type;
    return _this;
  }

  (0, _createClass2.default)(ResourceVariables, [{
    key: "all",
    value: function all(resourceId, resource2Id) {
      return _infrastructure.RequestHelper.get(this, url(this.resourceType, resourceId, this.resource2Type, resource2Id));
    }
  }, {
    key: "create",
    value: function create(resourceId, resource2Id, options) {
      return _infrastructure.RequestHelper.post(this, url(this.resourceType, resourceId, this.resource2Type, resource2Id), options);
    }
  }, {
    key: "edit",
    value: function edit(resourceId, resource2Id, keyId, options) {
      var kId = encodeURIComponent(keyId);
      return _infrastructure.RequestHelper.put(this, "".concat(url(this.resourceType, resourceId, this.resource2Type, resource2Id), "/").concat(kId), options);
    }
  }, {
    key: "show",
    value: function show(resourceId, resource2Id, keyId) {
      var kId = encodeURIComponent(keyId);
      return _infrastructure.RequestHelper.get(this, "".concat(url(this.resourceType, resourceId, this.resource2Type, resource2Id), "/").concat(kId));
    }
  }, {
    key: "remove",
    value: function remove(resourceId, resource2Id, keyId) {
      var kId = encodeURIComponent(keyId);
      return _infrastructure.RequestHelper.delete(this, "".concat(url(this.resourceType, resourceId, this.resource2Type, resource2Id), "/").concat(kId));
    }
  }]);
  return ResourceVariables;
}(_infrastructure.BaseService);

var _default = ResourceVariables;
exports.default = _default;
