// gitlab/dist/latest/templates/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ResourceAccessRequests", {
  enumerable: true,
  get: function () {
    return _ResourceAccessRequests.default;
  }
});
Object.defineProperty(exports, "ResourceAwardEmojis", {
  enumerable: true,
  get: function () {
    return _ResourceAwardEmojis.default;
  }
});
Object.defineProperty(exports, "ResourceBadges", {
  enumerable: true,
  get: function () {
    return _ResourceBadges.default;
  }
});
Object.defineProperty(exports, "ResourceCustomAttributes", {
  enumerable: true,
  get: function () {
    return _ResourceCustomAttributes.default;
  }
});
Object.defineProperty(exports, "ResourceDiscussions", {
  enumerable: true,
  get: function () {
    return _ResourceDiscussions.default;
  }
});
Object.defineProperty(exports, "ResourceIssueBoards", {
  enumerable: true,
  get: function () {
    return _ResourceIssueBoards.default;
  }
});
Object.defineProperty(exports, "ResourceMembers", {
  enumerable: true,
  get: function () {
    return _ResourceMembers.default;
  }
});
Object.defineProperty(exports, "ResourceMilestones", {
  enumerable: true,
  get: function () {
    return _ResourceMilestones.default;
  }
});
Object.defineProperty(exports, "ResourceNotes", {
  enumerable: true,
  get: function () {
    return _ResourceNotes.default;
  }
});
Object.defineProperty(exports, "ResourceTemplates", {
  enumerable: true,
  get: function () {
    return _ResourceTemplates.default;
  }
});
Object.defineProperty(exports, "ResourceVariables", {
  enumerable: true,
  get: function () {
    return _ResourceVariables.default;
  }
});

var _ResourceAccessRequests = _interopRequireDefault(require("./ResourceAccessRequests"));

var _ResourceAwardEmojis = _interopRequireDefault(require("./ResourceAwardEmojis"));

var _ResourceBadges = _interopRequireDefault(require("./ResourceBadges"));

var _ResourceCustomAttributes = _interopRequireDefault(require("./ResourceCustomAttributes"));

var _ResourceDiscussions = _interopRequireDefault(require("./ResourceDiscussions"));

var _ResourceIssueBoards = _interopRequireDefault(require("./ResourceIssueBoards"));

var _ResourceMembers = _interopRequireDefault(require("./ResourceMembers"));

var _ResourceMilestones = _interopRequireDefault(require("./ResourceMilestones"));

var _ResourceNotes = _interopRequireDefault(require("./ResourceNotes"));

var _ResourceTemplates = _interopRequireDefault(require("./ResourceTemplates"));

var _ResourceVariables = _interopRequireDefault(require("./ResourceVariables"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
