// gitlab/dist/es5/services/Branches.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var Branches =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(Branches, _BaseService);

  function Branches() {
    (0, _classCallCheck2.default)(this, Branches);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Branches).apply(this, arguments));
  }

  (0, _createClass2.default)(Branches, [{
    key: "all",
    value: function all(projectId, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/repository/branches"), options);
    }
  }, {
    key: "create",
    value: function create(projectId, branchName, ref) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/repository/branches"), {
        branch: branchName,
        ref: ref
      });
    }
  }, {
    key: "protect",
    value: function protect(projectId, branchName, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/protected_branches"), (0, _objectSpread2.default)({
        name: branchName
      }, options));
    }
  }, {
    key: "remove",
    value: function remove(projectId, branchName) {
      var _map = [projectId, branchName].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          pId = _map2[0],
          bName = _map2[1];

      return _infrastructure.RequestHelper.delete(this, "projects/".concat(pId, "/repository/branches/").concat(bName));
    }
  }, {
    key: "show",
    value: function show(projectId, branchName) {
      var _map3 = [projectId, branchName].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          pId = _map4[0],
          bName = _map4[1];

      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/repository/branches/").concat(bName));
    }
  }, {
    key: "unprotect",
    value: function unprotect(projectId, branchName) {
      var _map5 = [projectId, branchName].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 2),
          pId = _map6[0],
          bName = _map6[1];

      return _infrastructure.RequestHelper.put(this, "projects/".concat(pId, "/repository/branches/").concat(bName, "/unprotect"));
    }
  }]);
  return Branches;
}(_infrastructure.BaseService);

var _default = Branches;
exports.default = _default;
