// gitlab/dist/es5/services/BroadcastMessages.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var BroadcastMessages =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(BroadcastMessages, _BaseService);

  function BroadcastMessages() {
    (0, _classCallCheck2.default)(this, BroadcastMessages);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(BroadcastMessages).apply(this, arguments));
  }

  (0, _createClass2.default)(BroadcastMessages, [{
    key: "all",
    value: function all(options) {
      return _infrastructure.RequestHelper.get(this, 'broadcast_messages', options);
    }
  }, {
    key: "create",
    value: function create(options) {
      return _infrastructure.RequestHelper.post(this, 'broadcast_messages', options);
    }
  }, {
    key: "edit",
    value: function edit(broadcastMessageId, options) {
      var bId = encodeURIComponent(broadcastMessageId);
      return _infrastructure.RequestHelper.put(this, "broadcast_messages/".concat(bId), options);
    }
  }, {
    key: "remove",
    value: function remove(broadcastMessageId) {
      var bId = encodeURIComponent(broadcastMessageId);
      return _infrastructure.RequestHelper.delete(this, "broadcast_messages/".concat(bId));
    }
  }, {
    key: "show",
    value: function show(broadcastMessageId, options) {
      var bId = encodeURIComponent(broadcastMessageId);
      return _infrastructure.RequestHelper.get(this, "broadcast_messages/".concat(bId), options);
    }
  }]);
  return BroadcastMessages;
}(_infrastructure.BaseService);

var _default = BroadcastMessages;
exports.default = _default;
