// gitlab/dist/es5/services/Epics.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var Epics =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(Epics, _BaseService);

  function Epics() {
    (0, _classCallCheck2.default)(this, Epics);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Epics).apply(this, arguments));
  }

  (0, _createClass2.default)(Epics, [{
    key: "all",
    value: function all(groupId) {
      var gId = encodeURIComponent(groupId);
      return _infrastructure.RequestHelper.get(this, "groups/".concat(gId, "/epics"));
    }
  }, {
    key: "create",
    value: function create(groupId, title, options) {
      var gId = encodeURIComponent(groupId);
      return _infrastructure.RequestHelper.post(this, "groups/".concat(gId, "/epics"), (0, _objectSpread2.default)({
        title: title
      }, options));
    }
  }, {
    key: "edit",
    value: function edit(groupId, epicId, options) {
      var _map = [groupId, epicId].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          gId = _map2[0],
          eId = _map2[1];

      return _infrastructure.RequestHelper.put(this, "groups/".concat(gId, "/epics/").concat(eId), options);
    }
  }, {
    key: "remove",
    value: function remove(groupId, epicId) {
      var _map3 = [groupId, epicId].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          gId = _map4[0],
          eId = _map4[1];

      return _infrastructure.RequestHelper.delete(this, "groups/".concat(gId, "/epics/").concat(eId));
    }
  }, {
    key: "show",
    value: function show(groupId, epicId) {
      var _map5 = [groupId, epicId].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 2),
          gId = _map6[0],
          eId = _map6[1];

      return _infrastructure.RequestHelper.get(this, "groups/".concat(gId, "/epics/").concat(eId));
    }
  }]);
  return Epics;
}(_infrastructure.BaseService);

var _default = Epics;
exports.default = _default;
