// gitlab/dist/es5/services/GeoNodes.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var GeoNodes =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(GeoNodes, _BaseService);

  function GeoNodes() {
    (0, _classCallCheck2.default)(this, GeoNodes);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(GeoNodes).apply(this, arguments));
  }

  (0, _createClass2.default)(GeoNodes, [{
    key: "all",
    value: function all(options) {
      return _infrastructure.RequestHelper.get(this, 'geo_nodes', options);
    }
  }, {
    key: "create",
    value: function create(geonodeId, options) {
      var gId = encodeURIComponent(geonodeId);
      return _infrastructure.RequestHelper.post(this, "geo_nodes/".concat(gId), options);
    }
  }, {
    key: "edit",
    value: function edit(geonodeId, options) {
      var gId = encodeURIComponent(geonodeId);
      return _infrastructure.RequestHelper.put(this, "geo_nodes/".concat(gId), options);
    }
  }, {
    key: "failures",
    value: function failures(options) {
      return _infrastructure.RequestHelper.post(this, 'geo_nodes/current/failures', options);
    }
  }, {
    key: "repair",
    value: function repair(geonodeId, options) {
      var gId = encodeURIComponent(geonodeId);
      return _infrastructure.RequestHelper.delete(this, "geo_nodes/".concat(gId), options);
    }
  }, {
    key: "show",
    value: function show(geonodeId, options) {
      var gId = encodeURIComponent(geonodeId);
      return _infrastructure.RequestHelper.get(this, "geo_nodes/".concat(gId), options);
    }
  }, {
    key: "status",
    value: function status(geonodeId, options) {
      var gId = encodeURIComponent(geonodeId);
      return _infrastructure.RequestHelper.get(this, "geo_nodes/".concat(gId, "/status"), options);
    }
  }, {
    key: "statuses",
    value: function statuses(options) {
      return _infrastructure.RequestHelper.get(this, 'geo_nodes/statuses', options);
    }
  }]);
  return GeoNodes;
}(_infrastructure.BaseService);

var _default = GeoNodes;
exports.default = _default;
