// gitlab/dist/es5/services/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Groups", {
  enumerable: true,
  get: function get() {
    return _Groups.default;
  }
});
Object.defineProperty(exports, "GroupAccessRequests", {
  enumerable: true,
  get: function get() {
    return _GroupAccessRequests.default;
  }
});
Object.defineProperty(exports, "GroupBadges", {
  enumerable: true,
  get: function get() {
    return _GroupBadges.default;
  }
});
Object.defineProperty(exports, "GroupCustomAttributes", {
  enumerable: true,
  get: function get() {
    return _GroupCustomAttributes.default;
  }
});
Object.defineProperty(exports, "GroupIssueBoards", {
  enumerable: true,
  get: function get() {
    return _GroupIssueBoards.default;
  }
});
Object.defineProperty(exports, "GroupMembers", {
  enumerable: true,
  get: function get() {
    return _GroupMembers.default;
  }
});
Object.defineProperty(exports, "GroupMilestones", {
  enumerable: true,
  get: function get() {
    return _GroupMilestones.default;
  }
});
Object.defineProperty(exports, "GroupProjects", {
  enumerable: true,
  get: function get() {
    return _GroupProjects.default;
  }
});
Object.defineProperty(exports, "GroupVariables", {
  enumerable: true,
  get: function get() {
    return _GroupVariables.default;
  }
});
Object.defineProperty(exports, "Epics", {
  enumerable: true,
  get: function get() {
    return _Epics.default;
  }
});
Object.defineProperty(exports, "EpicIssues", {
  enumerable: true,
  get: function get() {
    return _EpicIssues.default;
  }
});
Object.defineProperty(exports, "EpicNotes", {
  enumerable: true,
  get: function get() {
    return _EpicNotes.default;
  }
});
Object.defineProperty(exports, "EpicDiscussions", {
  enumerable: true,
  get: function get() {
    return _EpicDiscussions.default;
  }
});
Object.defineProperty(exports, "Users", {
  enumerable: true,
  get: function get() {
    return _Users.default;
  }
});
Object.defineProperty(exports, "UserCustomAttributes", {
  enumerable: true,
  get: function get() {
    return _UserCustomAttributes.default;
  }
});
Object.defineProperty(exports, "UserEmails", {
  enumerable: true,
  get: function get() {
    return _UserEmails.default;
  }
});
Object.defineProperty(exports, "UserImpersonationTokens", {
  enumerable: true,
  get: function get() {
    return _UserImpersonationTokens.default;
  }
});
Object.defineProperty(exports, "UserKeys", {
  enumerable: true,
  get: function get() {
    return _UserKeys.default;
  }
});
Object.defineProperty(exports, "UserGPGKeys", {
  enumerable: true,
  get: function get() {
    return _UserGPGKeys.default;
  }
});
Object.defineProperty(exports, "Branches", {
  enumerable: true,
  get: function get() {
    return _Branches.default;
  }
});
Object.defineProperty(exports, "Commits", {
  enumerable: true,
  get: function get() {
    return _Commits.default;
  }
});
Object.defineProperty(exports, "CommitDiscussions", {
  enumerable: true,
  get: function get() {
    return _CommitDiscussions.default;
  }
});
Object.defineProperty(exports, "Deployments", {
  enumerable: true,
  get: function get() {
    return _Deployments.default;
  }
});
Object.defineProperty(exports, "DeployKeys", {
  enumerable: true,
  get: function get() {
    return _DeployKeys.default;
  }
});
Object.defineProperty(exports, "Environments", {
  enumerable: true,
  get: function get() {
    return _Environments.default;
  }
});
Object.defineProperty(exports, "Issues", {
  enumerable: true,
  get: function get() {
    return _Issues.default;
  }
});
Object.defineProperty(exports, "IssueNotes", {
  enumerable: true,
  get: function get() {
    return _IssueNotes.default;
  }
});
Object.defineProperty(exports, "IssueDiscussions", {
  enumerable: true,
  get: function get() {
    return _IssueDiscussions.default;
  }
});
Object.defineProperty(exports, "IssueAwardEmojis", {
  enumerable: true,
  get: function get() {
    return _IssueAwardEmojis.default;
  }
});
Object.defineProperty(exports, "Jobs", {
  enumerable: true,
  get: function get() {
    return _Jobs.default;
  }
});
Object.defineProperty(exports, "Labels", {
  enumerable: true,
  get: function get() {
    return _Labels.default;
  }
});
Object.defineProperty(exports, "MergeRequests", {
  enumerable: true,
  get: function get() {
    return _MergeRequests.default;
  }
});
Object.defineProperty(exports, "MergeRequestAwardEmojis", {
  enumerable: true,
  get: function get() {
    return _MergeRequestAwardEmojis.default;
  }
});
Object.defineProperty(exports, "MergeRequestDiscussions", {
  enumerable: true,
  get: function get() {
    return _MergeRequestDiscussions.default;
  }
});
Object.defineProperty(exports, "MergeRequestNotes", {
  enumerable: true,
  get: function get() {
    return _MergeRequestNotes.default;
  }
});
Object.defineProperty(exports, "Pipelines", {
  enumerable: true,
  get: function get() {
    return _Pipelines.default;
  }
});
Object.defineProperty(exports, "PipelineSchedules", {
  enumerable: true,
  get: function get() {
    return _PipelineSchedules.default;
  }
});
Object.defineProperty(exports, "PipelineScheduleVariables", {
  enumerable: true,
  get: function get() {
    return _PipelineScheduleVariables.default;
  }
});
Object.defineProperty(exports, "Projects", {
  enumerable: true,
  get: function get() {
    return _Projects.default;
  }
});
Object.defineProperty(exports, "ProjectAccessRequests", {
  enumerable: true,
  get: function get() {
    return _ProjectAccessRequests.default;
  }
});
Object.defineProperty(exports, "ProjectBadges", {
  enumerable: true,
  get: function get() {
    return _ProjectBadges.default;
  }
});
Object.defineProperty(exports, "ProjectCustomAttributes", {
  enumerable: true,
  get: function get() {
    return _ProjectCustomAttributes.default;
  }
});
Object.defineProperty(exports, "ProjectImportExport", {
  enumerable: true,
  get: function get() {
    return _ProjectImportExport.default;
  }
});
Object.defineProperty(exports, "ProjectIssueBoards", {
  enumerable: true,
  get: function get() {
    return _ProjectIssueBoards.default;
  }
});
Object.defineProperty(exports, "ProjectHooks", {
  enumerable: true,
  get: function get() {
    return _ProjectHooks.default;
  }
});
Object.defineProperty(exports, "ProjectMembers", {
  enumerable: true,
  get: function get() {
    return _ProjectMembers.default;
  }
});
Object.defineProperty(exports, "ProjectMilestones", {
  enumerable: true,
  get: function get() {
    return _ProjectMilestones.default;
  }
});
Object.defineProperty(exports, "ProjectSnippets", {
  enumerable: true,
  get: function get() {
    return _ProjectSnippets.default;
  }
});
Object.defineProperty(exports, "ProjectSnippetNotes", {
  enumerable: true,
  get: function get() {
    return _ProjectSnippetNotes.default;
  }
});
Object.defineProperty(exports, "ProjectSnippetDiscussions", {
  enumerable: true,
  get: function get() {
    return _ProjectSnippetDiscussions.default;
  }
});
Object.defineProperty(exports, "ProjectSnippetAwardEmojis", {
  enumerable: true,
  get: function get() {
    return _ProjectSnippetAwardEmojis.default;
  }
});
Object.defineProperty(exports, "ProtectedBranches", {
  enumerable: true,
  get: function get() {
    return _ProtectedBranches.default;
  }
});
Object.defineProperty(exports, "ProjectVariables", {
  enumerable: true,
  get: function get() {
    return _ProjectVariables.default;
  }
});
Object.defineProperty(exports, "Repositories", {
  enumerable: true,
  get: function get() {
    return _Repositories.default;
  }
});
Object.defineProperty(exports, "RepositoryFiles", {
  enumerable: true,
  get: function get() {
    return _RepositoryFiles.default;
  }
});
Object.defineProperty(exports, "Runners", {
  enumerable: true,
  get: function get() {
    return _Runners.default;
  }
});
Object.defineProperty(exports, "Services", {
  enumerable: true,
  get: function get() {
    return _Services.default;
  }
});
Object.defineProperty(exports, "Tags", {
  enumerable: true,
  get: function get() {
    return _Tags.default;
  }
});
Object.defineProperty(exports, "Todos", {
  enumerable: true,
  get: function get() {
    return _Todos.default;
  }
});
Object.defineProperty(exports, "Triggers", {
  enumerable: true,
  get: function get() {
    return _Triggers.default;
  }
});
Object.defineProperty(exports, "ApplicationSettings", {
  enumerable: true,
  get: function get() {
    return _ApplicationSettings.default;
  }
});
Object.defineProperty(exports, "BroadcastMessages", {
  enumerable: true,
  get: function get() {
    return _BroadcastMessages.default;
  }
});
Object.defineProperty(exports, "Events", {
  enumerable: true,
  get: function get() {
    return _Events.default;
  }
});
Object.defineProperty(exports, "FeatureFlags", {
  enumerable: true,
  get: function get() {
    return _FeatureFlags.default;
  }
});
Object.defineProperty(exports, "GeoNodes", {
  enumerable: true,
  get: function get() {
    return _GeoNodes.default;
  }
});
Object.defineProperty(exports, "GitignoreTemplates", {
  enumerable: true,
  get: function get() {
    return _GitignoreTemplates.default;
  }
});
Object.defineProperty(exports, "GitLabCIYMLTemplates", {
  enumerable: true,
  get: function get() {
    return _GitLabCIYMLTemplates.default;
  }
});
Object.defineProperty(exports, "Keys", {
  enumerable: true,
  get: function get() {
    return _Keys.default;
  }
});
Object.defineProperty(exports, "Licence", {
  enumerable: true,
  get: function get() {
    return _Licence.default;
  }
});
Object.defineProperty(exports, "LicenceTemplates", {
  enumerable: true,
  get: function get() {
    return _LicenceTemplates.default;
  }
});
Object.defineProperty(exports, "Lint", {
  enumerable: true,
  get: function get() {
    return _Lint.default;
  }
});
Object.defineProperty(exports, "Namespaces", {
  enumerable: true,
  get: function get() {
    return _Namespaces.default;
  }
});
Object.defineProperty(exports, "NotificationSettings", {
  enumerable: true,
  get: function get() {
    return _NotificationSettings.default;
  }
});
Object.defineProperty(exports, "PagesDomains", {
  enumerable: true,
  get: function get() {
    return _PagesDomains.default;
  }
});
Object.defineProperty(exports, "Search", {
  enumerable: true,
  get: function get() {
    return _Search.default;
  }
});
Object.defineProperty(exports, "SidekiqMetrics", {
  enumerable: true,
  get: function get() {
    return _SidekiqMetrics.default;
  }
});
Object.defineProperty(exports, "SystemHooks", {
  enumerable: true,
  get: function get() {
    return _SystemHooks.default;
  }
});
Object.defineProperty(exports, "Version", {
  enumerable: true,
  get: function get() {
    return _Version.default;
  }
});
Object.defineProperty(exports, "Wikis", {
  enumerable: true,
  get: function get() {
    return _Wikis.default;
  }
});

var _Groups = _interopRequireDefault(require("./Groups"));

var _GroupAccessRequests = _interopRequireDefault(require("./GroupAccessRequests"));

var _GroupBadges = _interopRequireDefault(require("./GroupBadges"));

var _GroupCustomAttributes = _interopRequireDefault(require("./GroupCustomAttributes"));

var _GroupIssueBoards = _interopRequireDefault(require("./GroupIssueBoards"));

var _GroupMembers = _interopRequireDefault(require("./GroupMembers"));

var _GroupMilestones = _interopRequireDefault(require("./GroupMilestones"));

var _GroupProjects = _interopRequireDefault(require("./GroupProjects"));

var _GroupVariables = _interopRequireDefault(require("./GroupVariables"));

var _Epics = _interopRequireDefault(require("./Epics"));

var _EpicIssues = _interopRequireDefault(require("./EpicIssues"));

var _EpicNotes = _interopRequireDefault(require("./EpicNotes"));

var _EpicDiscussions = _interopRequireDefault(require("./EpicDiscussions"));

var _Users = _interopRequireDefault(require("./Users"));

var _UserCustomAttributes = _interopRequireDefault(require("./UserCustomAttributes"));

var _UserEmails = _interopRequireDefault(require("./UserEmails"));

var _UserImpersonationTokens = _interopRequireDefault(require("./UserImpersonationTokens"));

var _UserKeys = _interopRequireDefault(require("./UserKeys"));

var _UserGPGKeys = _interopRequireDefault(require("./UserGPGKeys"));

var _Branches = _interopRequireDefault(require("./Branches"));

var _Commits = _interopRequireDefault(require("./Commits"));

var _CommitDiscussions = _interopRequireDefault(require("./CommitDiscussions"));

var _Deployments = _interopRequireDefault(require("./Deployments"));

var _DeployKeys = _interopRequireDefault(require("./DeployKeys"));

var _Environments = _interopRequireDefault(require("./Environments"));

var _Issues = _interopRequireDefault(require("./Issues"));

var _IssueNotes = _interopRequireDefault(require("./IssueNotes"));

var _IssueDiscussions = _interopRequireDefault(require("./IssueDiscussions"));

var _IssueAwardEmojis = _interopRequireDefault(require("./IssueAwardEmojis"));

var _Jobs = _interopRequireDefault(require("./Jobs"));

var _Labels = _interopRequireDefault(require("./Labels"));

var _MergeRequests = _interopRequireDefault(require("./MergeRequests"));

var _MergeRequestAwardEmojis = _interopRequireDefault(require("./MergeRequestAwardEmojis"));

var _MergeRequestDiscussions = _interopRequireDefault(require("./MergeRequestDiscussions"));

var _MergeRequestNotes = _interopRequireDefault(require("./MergeRequestNotes"));

var _Pipelines = _interopRequireDefault(require("./Pipelines"));

var _PipelineSchedules = _interopRequireDefault(require("./PipelineSchedules"));

var _PipelineScheduleVariables = _interopRequireDefault(require("./PipelineScheduleVariables"));

var _Projects = _interopRequireDefault(require("./Projects"));

var _ProjectAccessRequests = _interopRequireDefault(require("./ProjectAccessRequests"));

var _ProjectBadges = _interopRequireDefault(require("./ProjectBadges"));

var _ProjectCustomAttributes = _interopRequireDefault(require("./ProjectCustomAttributes"));

var _ProjectImportExport = _interopRequireDefault(require("./ProjectImportExport"));

var _ProjectIssueBoards = _interopRequireDefault(require("./ProjectIssueBoards"));

var _ProjectHooks = _interopRequireDefault(require("./ProjectHooks"));

var _ProjectMembers = _interopRequireDefault(require("./ProjectMembers"));

var _ProjectMilestones = _interopRequireDefault(require("./ProjectMilestones"));

var _ProjectSnippets = _interopRequireDefault(require("./ProjectSnippets"));

var _ProjectSnippetNotes = _interopRequireDefault(require("./ProjectSnippetNotes"));

var _ProjectSnippetDiscussions = _interopRequireDefault(require("./ProjectSnippetDiscussions"));

var _ProjectSnippetAwardEmojis = _interopRequireDefault(require("./ProjectSnippetAwardEmojis"));

var _ProtectedBranches = _interopRequireDefault(require("./ProtectedBranches"));

var _ProjectVariables = _interopRequireDefault(require("./ProjectVariables"));

var _Repositories = _interopRequireDefault(require("./Repositories"));

var _RepositoryFiles = _interopRequireDefault(require("./RepositoryFiles"));

var _Runners = _interopRequireDefault(require("./Runners"));

var _Services = _interopRequireDefault(require("./Services"));

var _Tags = _interopRequireDefault(require("./Tags"));

var _Todos = _interopRequireDefault(require("./Todos"));

var _Triggers = _interopRequireDefault(require("./Triggers"));

var _ApplicationSettings = _interopRequireDefault(require("./ApplicationSettings"));

var _BroadcastMessages = _interopRequireDefault(require("./BroadcastMessages"));

var _Events = _interopRequireDefault(require("./Events"));

var _FeatureFlags = _interopRequireDefault(require("./FeatureFlags"));

var _GeoNodes = _interopRequireDefault(require("./GeoNodes"));

var _GitignoreTemplates = _interopRequireDefault(require("./GitignoreTemplates"));

var _GitLabCIYMLTemplates = _interopRequireDefault(require("./GitLabCIYMLTemplates"));

var _Keys = _interopRequireDefault(require("./Keys"));

var _Licence = _interopRequireDefault(require("./Licence"));

var _LicenceTemplates = _interopRequireDefault(require("./LicenceTemplates"));

var _Lint = _interopRequireDefault(require("./Lint"));

var _Namespaces = _interopRequireDefault(require("./Namespaces"));

var _NotificationSettings = _interopRequireDefault(require("./NotificationSettings"));

var _PagesDomains = _interopRequireDefault(require("./PagesDomains"));

var _Search = _interopRequireDefault(require("./Search"));

var _SidekiqMetrics = _interopRequireDefault(require("./SidekiqMetrics"));

var _SystemHooks = _interopRequireDefault(require("./SystemHooks"));

var _Version = _interopRequireDefault(require("./Version"));

var _Wikis = _interopRequireDefault(require("./Wikis"));
