// gitlab/dist/es5/services/Issues.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var Issues =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(Issues, _BaseService);

  function Issues() {
    (0, _classCallCheck2.default)(this, Issues);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Issues).apply(this, arguments));
  }

  (0, _createClass2.default)(Issues, [{
    key: "addSpentTime",
    value: function addSpentTime(projectId, issueId, duration) {
      var _map = [projectId, issueId].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          pId = _map2[0],
          iId = _map2[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(iId, "/add_spent_time"), {
        duration: duration
      });
    }
  }, {
    key: "addTimeEstimate",
    value: function addTimeEstimate(projectId, issueId, duration) {
      var _map3 = [projectId, issueId].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          pId = _map4[0],
          iId = _map4[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(iId, "/time_estimate"), {
        duration: duration
      });
    }
  }, {
    key: "all",
    value: function all(_ref) {
      var projectId = _ref.projectId,
          options = (0, _objectWithoutProperties2.default)(_ref, ["projectId"]);
      var url = projectId ? "projects/".concat(encodeURIComponent(projectId), "/issues") : 'issues';
      return _infrastructure.RequestHelper.get(this, url, options);
    }
  }, {
    key: "create",
    value: function create(projectId, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues"), options);
    }
  }, {
    key: "edit",
    value: function edit(projectId, issueId, options) {
      var _map5 = [projectId, issueId].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 2),
          pId = _map6[0],
          iId = _map6[1];

      return _infrastructure.RequestHelper.put(this, "projects/".concat(pId, "/issues/").concat(iId), options);
    }
  }, {
    key: "link",
    value: function link(projectId, issueIId, targetProjectId, targetIssueId) {
      var options = arguments.length > 4 && arguments[4] !== undefined ? arguments[4] : {};

      var _map7 = [projectId, issueIId].map(encodeURIComponent),
          _map8 = (0, _slicedToArray2.default)(_map7, 2),
          pId = _map8[0],
          iId = _map8[1];

      var _map9 = [targetProjectId, targetIssueId].map(encodeURIComponent),
          _map10 = (0, _slicedToArray2.default)(_map9, 2),
          targetpId = _map10[0],
          targetIId = _map10[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(iId, "/links"), (0, _objectSpread2.default)({
        targetProjectId: targetpId,
        targetIssueId: targetIId
      }, options));
    }
  }, {
    key: "participants",
    value: function participants(projectId, issueId) {
      var _map11 = [projectId, issueId].map(encodeURIComponent),
          _map12 = (0, _slicedToArray2.default)(_map11, 2),
          pId = _map12[0],
          iId = _map12[1];

      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/issues/").concat(iId, "/participants"));
    }
  }, {
    key: "remove",
    value: function remove(projectId, issueId) {
      var _map13 = [projectId, issueId].map(encodeURIComponent),
          _map14 = (0, _slicedToArray2.default)(_map13, 2),
          pId = _map14[0],
          iId = _map14[1];

      return _infrastructure.RequestHelper.delete(this, "projects/".concat(pId, "/issues/").concat(iId));
    }
  }, {
    key: "resetSpentTime",
    value: function resetSpentTime(projectId, mergerequestId) {
      var _map15 = [projectId, mergerequestId].map(encodeURIComponent),
          _map16 = (0, _slicedToArray2.default)(_map15, 2),
          pId = _map16[0],
          mId = _map16[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(mId, "/reset_spent_time"));
    }
  }, {
    key: "resetTimeEstimate",
    value: function resetTimeEstimate(projectId, mergerequestId) {
      var _map17 = [projectId, mergerequestId].map(encodeURIComponent),
          _map18 = (0, _slicedToArray2.default)(_map17, 2),
          pId = _map18[0],
          mId = _map18[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(mId, "/reset_time_estimate"));
    }
  }, {
    key: "show",
    value: function show(projectId, issueId) {
      var _map19 = [projectId, issueId].map(encodeURIComponent),
          _map20 = (0, _slicedToArray2.default)(_map19, 2),
          pId = _map20[0],
          iId = _map20[1];

      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/issues/").concat(iId));
    }
  }, {
    key: "subscribe",
    value: function subscribe(projectId, issueId, options) {
      var _map21 = [projectId, issueId].map(encodeURIComponent),
          _map22 = (0, _slicedToArray2.default)(_map21, 2),
          pId = _map22[0],
          iId = _map22[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/issues/").concat(iId, "/subscribe"), options);
    }
  }, {
    key: "timeStats",
    value: function timeStats(projectId, mergerequestId) {
      var _map23 = [projectId, mergerequestId].map(encodeURIComponent),
          _map24 = (0, _slicedToArray2.default)(_map23, 2),
          pId = _map24[0],
          mId = _map24[1];

      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/issues/").concat(mId, "/time_stats"));
    }
  }, {
    key: "unsubscribe",
    value: function unsubscribe(projectId, issueId) {
      var _map25 = [projectId, issueId].map(encodeURIComponent),
          _map26 = (0, _slicedToArray2.default)(_map25, 2),
          pId = _map26[0],
          iId = _map26[1];

      return _infrastructure.RequestHelper.delete(this, "projects/".concat(pId, "/issues/").concat(iId, "/unsubscribe"));
    }
  }]);
  return Issues;
}(_infrastructure.BaseService);

var _default = Issues;
exports.default = _default;
