// gitlab/dist/es5/services/NotificationSettings.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf3 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var LEVELS = {
  DISABLED: 'disabled',
  PARTICIPATING: 'participating',
  WATCH: 'watch',
  GLOBAL: 'global',
  MENTION: 'mention',
  CUSTOM: 'custom'
};
var EVENTS = {
  NEW_NOTE: 'new_note',
  NEW_ISSUE: 'new_issue',
  REOPEM_ISSUE: 'reopen_issue',
  CLOSE_ISSUE: 'close_issue',
  REASSIGN_ISSUE: 'reassign_issue',
  NEW_MERGE_REQUESTS: 'new_merge_request',
  PUSH_TO_MERGE_REQUEST: 'push_to_merge_request',
  REOPEN_MERGE_REQUESTS: 'reopen_merge_request',
  CLOSE_MERGE_REQUEST: 'close_merge_request',
  REASSIGN_MERGE_REQUEST: 'reassign_merge_request',
  MERGE_MERGE_REQUEST: 'merge_merge_request',
  FAILED_PIPELINE: 'failed_pipeline',
  SUCCESS_PIPELINE: 'success_pipeline'
};

var NotificationSettings =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(NotificationSettings, _BaseService);

  function NotificationSettings() {
    var _getPrototypeOf2;

    var _this;

    (0, _classCallCheck2.default)(this, NotificationSettings);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = (0, _possibleConstructorReturn2.default)(this, (_getPrototypeOf2 = (0, _getPrototypeOf3.default)(NotificationSettings)).call.apply(_getPrototypeOf2, [this].concat(args)));
    _this.LEVELS = LEVELS;
    _this.EVENTS = EVENTS;
    return _this;
  }

  (0, _createClass2.default)(NotificationSettings, [{
    key: "all",
    value: function all() {
      var _ref = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {},
          projectId = _ref.projectId,
          groupId = _ref.groupId;

      var url;

      if (projectId) {
        url += "projects/".concat(encodeURIComponent(projectId), "/");
      } else if (groupId) {
        url += "groups/".concat(encodeURIComponent(groupId), "/");
      }

      return _infrastructure.RequestHelper.get(this, "".concat(url, "notification_settings"));
    }
  }, {
    key: "edit",
    value: function edit(options) {
      var _ref2 = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {},
          projectId = _ref2.projectId,
          groupId = _ref2.groupId;

      var url;

      if (projectId) {
        url += "projects/".concat(encodeURIComponent(projectId), "/");
      } else if (groupId) {
        url += "groups/".concat(encodeURIComponent(groupId), "/");
      }

      return _infrastructure.RequestHelper.put(this, "".concat(url, "notification_settings"), options);
    }
  }]);
  return NotificationSettings;
}(_infrastructure.BaseService);

var _default = NotificationSettings;
exports.default = _default;
