// gitlab/dist/es5/services/PipelineSchedules.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread"));

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var PipelineSchedules =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(PipelineSchedules, _BaseService);

  function PipelineSchedules() {
    (0, _classCallCheck2.default)(this, PipelineSchedules);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(PipelineSchedules).apply(this, arguments));
  }

  (0, _createClass2.default)(PipelineSchedules, [{
    key: "all",
    value: function all(projectId, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/pipeline_schedules"), options);
    }
  }, {
    key: "create",
    value: function create(projectId, description, ref, cron, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/pipeline_schedules"), (0, _objectSpread2.default)({
        description: description,
        ref: ref,
        cron: cron
      }, options));
    }
  }, {
    key: "edit",
    value: function edit(projectId, scheduleId, options) {
      var _map = [projectId, scheduleId].map(encodeURIComponent),
          _map2 = (0, _slicedToArray2.default)(_map, 2),
          pId = _map2[0],
          sId = _map2[1];

      return _infrastructure.RequestHelper.put(this, "projects/".concat(pId, "/pipeline_schedules/").concat(sId), options);
    }
  }, {
    key: "remove",
    value: function remove(projectId, scheduleId) {
      var _map3 = [projectId, scheduleId].map(encodeURIComponent),
          _map4 = (0, _slicedToArray2.default)(_map3, 2),
          pId = _map4[0],
          sId = _map4[1];

      return _infrastructure.RequestHelper.delete(this, "projects/".concat(pId, "/pipeline_schedules/").concat(sId));
    }
  }, {
    key: "show",
    value: function show(projectId, scheduleId) {
      var _map5 = [projectId, scheduleId].map(encodeURIComponent),
          _map6 = (0, _slicedToArray2.default)(_map5, 2),
          pId = _map6[0],
          sId = _map6[1];

      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/pipeline_schedules/").concat(sId));
    }
  }, {
    key: "takeOwnership",
    value: function takeOwnership(projectId, scheduleId) {
      var _map7 = [projectId, scheduleId].map(encodeURIComponent),
          _map8 = (0, _slicedToArray2.default)(_map7, 2),
          pId = _map8[0],
          sId = _map8[1];

      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/pipeline_schedules/").concat(sId, "/take_ownership"));
    }
  }]);
  return PipelineSchedules;
}(_infrastructure.BaseService);

var _default = PipelineSchedules;
exports.default = _default;
