// gitlab/dist/es5/services/Wikis.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _infrastructure = require("../infrastructure");

var Wikis =
/*#__PURE__*/
function (_BaseService) {
  (0, _inherits2.default)(Wikis, _BaseService);

  function Wikis() {
    (0, _classCallCheck2.default)(this, Wikis);
    return (0, _possibleConstructorReturn2.default)(this, (0, _getPrototypeOf2.default)(Wikis).apply(this, arguments));
  }

  (0, _createClass2.default)(Wikis, [{
    key: "all",
    value: function all(projectId, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/wikis"), options);
    }
  }, {
    key: "create",
    value: function create(projectId, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.post(this, "projects/".concat(pId, "/wikis"), options);
    }
  }, {
    key: "edit",
    value: function edit(projectId, slug, options) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.put(this, "projects/".concat(pId, "/wikis/").concat(slug), options);
    }
  }, {
    key: "show",
    value: function show(projectId, slug) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.get(this, "projects/".concat(pId, "/wikis/").concat(slug));
    }
  }, {
    key: "remove",
    value: function remove(projectId, slug) {
      var pId = encodeURIComponent(projectId);
      return _infrastructure.RequestHelper.delete(this, "projects/".concat(pId, "/wikis/").concat(slug));
    }
  }]);
  return Wikis;
}(_infrastructure.BaseService);

var _default = Wikis;
exports.default = _default;
