// gitlab/dist/latest/services/Branches.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _infrastructure = require("../infrastructure");

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

class Branches extends _infrastructure.BaseService {
  all(projectId, options) {
    const pId = encodeURIComponent(projectId);
    return _infrastructure.RequestHelper.get(this, `projects/${pId}/repository/branches`, options);
  }

  create(projectId, branchName, ref) {
    const pId = encodeURIComponent(projectId);
    return _infrastructure.RequestHelper.post(this, `projects/${pId}/repository/branches`, {
      branch: branchName,
      ref
    });
  }

  protect(projectId, branchName, options) {
    const pId = encodeURIComponent(projectId);
    return _infrastructure.RequestHelper.post(this, `projects/${pId}/protected_branches`, _objectSpread({
      name: branchName
    }, options));
  }

  remove(projectId, branchName) {
    const [pId, bName] = [projectId, branchName].map(encodeURIComponent);
    return _infrastructure.RequestHelper.delete(this, `projects/${pId}/repository/branches/${bName}`);
  }

  show(projectId, branchName) {
    const [pId, bName] = [projectId, branchName].map(encodeURIComponent);
    return _infrastructure.RequestHelper.get(this, `projects/${pId}/repository/branches/${bName}`);
  }

  unprotect(projectId, branchName) {
    const [pId, bName] = [projectId, branchName].map(encodeURIComponent);
    return _infrastructure.RequestHelper.put(this, `projects/${pId}/repository/branches/${bName}/unprotect`);
  }

}

var _default = Branches;
exports.default = _default;
